numpy/core/function_base.py

```diff
@@ -1,10 +1,24 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
+def _index_deprecate(i, stacklevel=2):
+    try:
+        i = operator.index(i)
+    except TypeError:
+        msg = ("object of type {} cannot be safely interpreted as "
+               "an integer.".format(type(i)))
+        stacklevel += 1
```

Comment:
That's correct -- the +1 corrects for the helper function's stack frame, and then stacklevel 2 means to warn the caller of linspace (rather than warning linspace itself).